httpcore/_async/connection.py

```diff
@@ -70,16 +77,23 @@ def request_lock(self) -> AsyncLock:
         scheme, hostname, port = self.origin
         timeout = {} if timeout is None else timeout
         ssl_context = self.ssl_context if scheme == b"https" else None
-        socket = await self.backend.open_tcp_stream(
+        self.socket = await self.backend.open_tcp_stream(
             hostname, port, ssl_context, timeout
         )
+        self._create_connection(self.socket)
+
+    def _create_connection(self, socket: AsyncSocketStream) -> None:
         http_version = socket.get_http_version()
         if http_version == "HTTP/2":
             self.is_http2 = True
-            self.connection = AsyncHTTP2Connection(socket=socket, backend=self.backend)
+            self.connection = AsyncHTTP2Connection(
+                socket=socket, backend=self.backend, ssl_context=self.ssl_context
```

Comment:
Is the lack of passing `self.ssl_context` here an unrelated possible source of bug? Maybe we can submit this as a separate PR…

Reply:
It is the source of a upcoming bug in the HTTPS handling, and very likely a source of other undetected bugs.

I'd prefer not to open another PR just for this though, if you don't mind 🙂

(Sorry for the double posting, I replied to the wrong comment 😅 )

Reply:
> Maybe we can submit this as a separate PR…

Yeah, let's drop that idea